integration/deltaspike-data/impl/src/main/java/com/blazebit/persistence/impl/deltaspikedata/impl/meta/extractor/EntityViewTypeMetadataExtractor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.deltaspikedata.impl.meta.extractor;
+
+import com.blazebit.persistence.impl.deltaspikedata.impl.meta.RepositoryEntityView;
+import com.blazebit.persistence.view.EntityView;
+
+import java.io.Serializable;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Implementation is similar to {@link org.apache.deltaspike.data.impl.meta.extractor.TypeMetadataExtractor} but was modified to
+ * work with entity views.
+ *
+ * @author Moritz Becker
+ * @since 1.2.0
+ */
+public class EntityViewTypeMetadataExtractor implements EntityViewMetadataExtractor {
+    private static final Logger LOG = Logger.getLogger(EntityViewTypeMetadataExtractor.class.getName());
+
+    @Override
+    public RepositoryEntityView extract(Class<?> repoClass) {
+        for (Type inf : repoClass.getGenericInterfaces()) {
+            RepositoryEntityView result = extractFrom(inf);
+            if (result != null) {
+                return result;
+            }
+        }
+        RepositoryEntityView result = extractFrom(repoClass.getGenericSuperclass());
+        if (result != null) {
+            return result;
+        }
+        for (Type intf : repoClass.getGenericInterfaces()) {
+            result = extractFrom(intf);
+            if (result != null) {
+                return result;
+            }
+        }
+        if (repoClass.getSuperclass() != null) {
+            return extract(repoClass.getSuperclass());
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    private RepositoryEntityView extractFrom(Type type) {
+        LOG.log(Level.FINER, "extractFrom: type = {0}", type);
+        if (!(type instanceof ParameterizedType)) {
+            return null;
+        }
+
+        ParameterizedType parametrizedType = (ParameterizedType) type;
+        Type[] genericTypes = parametrizedType.getActualTypeArguments();
+
+        RepositoryEntityView result = null;
+
+        // don't use a foreach here, we must be sure that the we first get the entity type
+        for (int i = 0; i < genericTypes.length; i++) {
+            Type genericType = genericTypes[i];
+
+            if (genericType instanceof Class && ((Class<?>) genericType).isAnnotationPresent(EntityView.class)) {
```

Comment:
The same comment I gave for `EntityViewAnnotationMetadataExtractor` applies here, using the `EntityViewManager` would be much better here.

Reply:
Not possible since this happens within ProcessAnnotatedType in the CDI extension and the EntityViewManager is not available at that time.